legacy/packetinterfacetcpserver.cpp

```diff
@@ -19,167 +24,199 @@ PacketInterfaceTCPServer::PacketInterfaceTCPServer(QObject *parent) : QObject(pa
+                    //                ret.vbAppendDouble32(mVehicleState->getPosition(PosType::GNSS).getYaw(), 1e6); // yaw in degrees
+                    ret.vbAppendDouble32(mVehicleState->getPosition().getRoll(), 1e6);
+                    ret.vbAppendDouble32(mVehicleState->getPosition().getPitch(), 1e6);
+                    ret.vbAppendDouble32(mVehicleState->getPosition().getYaw()*180.0/M_PI, 1e6); // yaw in degrees
+                    ret.vbAppendDouble32(mVehicleState->getAccelerometerXYZ()[0], 1e6); // accel_x in g
+                    ret.vbAppendDouble32(mVehicleState->getAccelerometerXYZ()[1], 1e6); // accel_y in g
+                    ret.vbAppendDouble32(mVehicleState->getAccelerometerXYZ()[2], 1e6); // accel_z in g
+                    ret.vbAppendDouble32(mVehicleState->getGyroscopeXYZ()[0]*M_PI/180.0, 1e6); // roll_rate in radians per second
+                    ret.vbAppendDouble32(mVehicleState->getGyroscopeXYZ()[1]*M_PI/180.0, 1e6); // pitch_rate in radians per second
+                    ret.vbAppendDouble32(mVehicleState->getGyroscopeXYZ()[2]*M_PI/180.0, 1e6); // yaw_rate in radians per second
+                    ret.vbAppendDouble32(0.0, 1e6); // magnet_x
+                    ret.vbAppendDouble32(0.0, 1e6); // magnet_y
+                    ret.vbAppendDouble32(0.0, 1e6); // magnet_z
+                    ret.vbAppendDouble32(mVehicleState->getPosition(PosType::GNSS).getX(), 1e4); // Should be PosType::fused
+                    ret.vbAppendDouble32(mVehicleState->getPosition(PosType::GNSS).getY(), 1e4); // Should be PosType::fused
+                    ret.vbAppendDouble32(mVehicleState->getSpeed(), 1e6);
+                    ret.vbAppendDouble32(-1.0, 1e6); // v_in
+                    ret.vbAppendDouble32(-1.0, 1e6); // temp mos
+                    ret.vbAppendUint8(0); // MC Fault code
+                    ret.vbAppendDouble32(mVehicleState->getPosition(PosType::GNSS).getX(), 1e4);
+                    ret.vbAppendDouble32(mVehicleState->getPosition(PosType::GNSS).getY(), 1e4);
+                    ret.vbAppendDouble32(wpFollowerState.currentGoal.getX(), 1e4); // autopilot_goal_x
+                    ret.vbAppendDouble32(wpFollowerState.currentGoal.getY(), 1e4); // autopilot_goal_y
+                    ret.vbAppendDouble32(wpFollowerState.purePursuitRadius, 1e6); // autopilot_pp_radius
+                    ret.vbAppendInt32(QTime::currentTime().addSecs(-QDateTime::currentDateTime().offsetFromUtc()).msecsSinceStartOfDay());
+                    ret.vbAppendInt16(0); // autopilot_route_left
+                    ret.vbAppendDouble32(mVehicleState->getPosition(PosType::UWB).getX(), 1e4); // UWB px
+                    ret.vbAppendDouble32(mVehicleState->getPosition(PosType::UWB).getY(), 1e4); // UWB PY
+                    mTcpServer.packet()->sendPacket(ret);
                 }
+            } break;
+                // --- Set state on vehicle
+            case CMD_SET_POS:
```

Comment:
this needs to go directly above CMD_SET_POS_ACK and the case needs to be changed like it was before: CMD_SET_POS and CMD_SET_POS_ACK do exactly the same, if it was actually CMD_SET_POS_ACK then ack is sent back

Reply:
Why do we want to use case for acks here? The acks are sent back to base station and are handled there. Before it was like this:

        // --- Set state on vehicle
        case CMD_SET_POS:
        case CMD_SET_POS_ACK: {
            PosPoint tmpPos = mVehicleState->getPosition();
            tmpPos.setX(packetData.vbPopFrontDouble32(1e4));
            tmpPos.setY(packetData.vbPopFrontDouble32(1e4));
            tmpPos.setYaw(packetData.vbPopFrontDouble32(1e6) * M_PI/180.0);
            mVehicleState->setPosition(tmpPos);

            if (commandID == CMD_SET_POS_ACK) {
                VByteArray ack;
                ack.vbAppendUint8(mVehicleState->getId());
                ack.vbAppendUint8(commandID);
                mTcpServer.packet()->sendPacket(ack);
            }
        } break;
        // --- Remote control vehicle